src/stream.ts

```diff
@@ -77,25 +60,34 @@ export class YamuxStream implements Stream {
   private epochStart: number
   private readonly getRTT: () => number
 
-  /** Used to stop the sink */
-  private readonly abortController: AbortController
-
   private readonly sendFrame: (header: FrameHeader, body?: Uint8Array) => void
-  private readonly onStreamEnd: () => void
 
   constructor (init: YamuxStreamInit) {
+    super({
+      ...init,
+      onEnd: (err?: Error) => {
+        this.state = StreamState.Finished
+        init.onEnd?.(err)
+      },
+      onCloseRead: () => {
+        this.readState = HalfStreamState.Closed
+      },
+      onCloseWrite: () => {
+        this.writeState = HalfStreamState.Closed
+      },
+      onReset: () => {
+        this.readState = HalfStreamState.Reset
+        this.writeState = HalfStreamState.Reset
+      },
+      onAbort: () => {
+        this.readState = HalfStreamState.Reset
+        this.writeState = HalfStreamState.Reset
```

Comment:
it seems before we were checking the read/write states before overwriting them.
Eg: if `(this.readState === Open) this.readState = Reset`
Not sure if that is still necessary or not

Reply:
I don't think it's necessary since aborting a stream closes if for reading and writing.

I guess you could close the read end (for example) then abort but I don't know if this is a big deal or not.

The super class has `.readStatus`/`.writeStatus` so maybe these fields can be removed?

Reply:
Yeah, they don't seem to be used anywhere now

Reply:
I've removed them and updated the tests to assert on `.readStatus`/`.writeStatus` instead.